### PR TITLE
[test-integration] Small cleanups on integration cli

### PR DESCRIPTION
**Pull request by vdemeester**
- Join a few tests in one when it makes sense (reduce the number of container run and thus the overall time of the suites)
- Remove some duplication on several tests
- Remove some unused methods

/cc @thaJeztah @icecrime @AkihiroSuda @dnephin @cpuguy83 @tiborvass 

🐸

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

### CODE REVIEW

File: integration-cli/trust_server_test.go [hunk 1]
```diff
@@ -234,24 +235,24 @@ func (s *DockerTrustSuite) setupTrustedplugin(c *check.C, source, name string) s
...
 		fmt.Sprintf("NOTARY_SNAPSHOT_PASSPHRASE=%s", pwd),
 		fmt.Sprintf("NOTARY_DELEGATION_PASSPHRASE=%s", pwd),
 	}
-	cmd.Env = append(os.Environ(), env...)
+	result := icmd.RunCmd(icmd.Cmd{
+		Command: append([]string{notaryBinary, "-c", filepath.Join(s.not.dir, "client-config.json")}, args...),
+		Env:     append(os.Environ(), env...),
+	})
+	result.Assert(c, icmd.Success)
+	return result.Combined()
```
Comment (dnephin) [inline integration-cli/trust_server_test.go]: :heart:

File: integration-cli/docker_experimental_network_test.go [hunk 2]
```diff
@@ -13,29 +13,23 @@ import (
...
+}
+
+func checkKernelMajorVersionGreaterOrEqualThen(kernelVersion int, majorVersion int) bool {
 	kv, err := kernel.GetKernelVersion()
 	if err != nil {
 		return false
 	}
-	// ensure Kernel version is >= v4.2 for ipvlan support
-	if kv.Kernel < ipvlanKernelVer || (kv.Kernel == ipvlanKernelVer && kv.Major < ipvlanMajorVer) {
+	// ensure Kernel version is >= v3.9 for macvlan support
```
Comment (dnephin) [inline integration-cli/docker_experimental_network_test.go]: Minor: this comment could be updated

File: integration-cli/docker_cli_service_health_test.go [hunk 3]
```diff
@@ -132,60 +132,3 @@ func (s *DockerSwarmSuite) TestServiceHealthStart(c *check.C) {
 		return task.Status.State, nil
 	}, checker.Equals, swarm.TaskStateRunning)
 }
-
-// start a service whose task is unhealthy at beginning
-// its tasks should be blocked in starting stage, until health check is passed
-func (s *DockerSwarmSuite) TestServiceHealthUpdate(c *check.C) {
```
Comment (dnephin) [inline integration-cli/docker_cli_service_health_test.go]: What's the story with this test?

Comment (vdemeester) [inline integration-cli/docker_cli_service_health_test.go]: No idea, but the diff with the previous one is just an empty line so…
ping @runshenzhu

Comment (dnephin) [inline integration-cli/docker_cli_service_health_test.go]: Ahh, I see. It's a duplicate.

I suspect it should be testing something else, but the line wasn't updated after the test was copied.

File: integration-cli/docker_cli_logs_test.go [hunk 4]
```diff
@@ -16,40 +16,24 @@ import (
...
 }
 
 // Regression test: When going much over the PageSize, it used to block (gh#4851)
 func (s *DockerSuite) TestLogsContainerMuchBiggerThanPage(c *check.C) {
-	testLen := 33000
-	out, _ := dockerCmd(c, "run", "-d", "busybox", "sh", "-c", fmt.Sprintf("for i in $(seq 1 %d); do echo -n = >> a.a; done; echo >> a.a; cat a.a", testLen))
+	testLogsContainerPagezie(c, 33000)
+}
 
+func testLogsContainerPagezie(c *check.C, testLen int) {
```
Comment (thaJeztah) [inline integration-cli/docker_cli_logs_test.go]: `Pagezie`? `PageSize`?

File: integration-cli/docker_cli_run_test.go [hunk 5]
```diff
@@ -1858,13 +1858,20 @@ func (s *DockerSuite) TestRunInteractiveWithRestartPolicy(c *check.C) {
 }
 
 // Test for #2267
-func (s *DockerSuite) TestRunWriteHostsFileAndNotCommit(c *check.C) {
+func (s *DockerSuite) TestRunWriteSpecialFilesAndNotCommit(c *check.C) {
 	// Cannot run on Windows as Windows does not support diff.
 	testRequires(c, DaemonIsLinux)
```
Comment (thaJeztah) [inline integration-cli/docker_cli_run_test.go]: Actually; https://github.com/docker/docker/pull/27842

File: integration-cli/docker_cli_run_test.go [hunk 6]
```diff
@@ -1901,38 +1908,6 @@ func sliceEq(a, b []string) bool {
...
-
-	out, _ = dockerCmd(c, "diff", name)
-	if len(strings.Trim(out, "\r\n")) != 0 && !eqToBaseDiff(out, c) {
-		c.Fatal("diff should be empty")
-	}
-}
-
 func (s *DockerSuite) TestRunWithBadDevice(c *check.C) {
 	// Cannot run on Windows as Windows does not support --device
 	testRequires(c, DaemonIsLinux)
```
Comment (thaJeztah) [inline integration-cli/docker_cli_run_test.go]: Same here, possibly https://github.com/docker/docker/pull/27842

Comment (thaJeztah) [inline integration-cli/docker_cli_run_test.go]: Although `cgroups` probably doesn't make a lot of sense for Windows 😄

File: integration-cli/docker_cli_run_test.go [hunk 7]
```diff
@@ -4519,31 +4433,17 @@ func (s *DockerSuite) TestRunDuplicateMount(c *check.C) {
 	c.Assert(out, checker.Contains, "null")
 }
 
-func (s *DockerSuite) TestRunWindowsWithCPUCount(c *check.C) {
+func (s *DockerSuite) TestRunWindowsWithCPUCountCPUSharesAndCPUPercent(c *check.C) {
 	testRequires(c, DaemonIsWindows)
 
-	out, _ := dockerCmd(c, "run", "--cpu-count=1", "--name", "test", "busybox", "echo", "testing")
+	out, _ := dockerCmd(c, "run", "--cpu-percent=80", "--cpu-count=1", "--cpu-shares=1000", "--name", "test", "busybox", "echo", "testing")
```
Comment (thaJeztah) [inline integration-cli/docker_cli_run_test.go]: Can these actually be combined? https://github.com/docker/docker/blob/9fab262690a1e9c465687b3dc22f84b053c5624f/daemon/daemon_windows.go#L89-L128

**Review by thaJeztah — CHANGES_REQUESTED** [review]
left some questions / suggestions

File: integration-cli/docker_cli_logs_test.go (hunk 4 above)
Comment (aaronlehmann) [inline integration-cli/docker_cli_logs_test.go]: `Pagination`?

File: integration-cli/docker_cli_service_health_test.go (hunk 3 above)
Comment (runshenzhu) [inline integration-cli/docker_cli_service_health_test.go]: @vdemeester it seems that I made a mistake when writing it. Very sorry about it.

Comment (vdemeester) [inline integration-cli/docker_cli_service_health_test.go]: @runshenzhu no problem 😉

File: integration-cli/docker_cli_run_test.go (hunk 7 above)
Comment (vdemeester) [inline integration-cli/docker_cli_run_test.go]: arf, I am sad 😓

**Review by thaJeztah — APPROVED** [review]
LGTM

**Review by cpuguy83 — APPROVED** [review]
LGTM

**Pull request merged**